enhancements/machine-api/cluster-api-integration.md

```diff
@@ -0,0 +1,269 @@
+
+##### Create user data secret
+
+Cluster API Machines will need a user data secret, similar to the one that Machine API uses.
+This secret is created by installer for Machine API.
+While Cluster API components are in technical preview, and therefore not integrated into the OpenShift Installer, the operator can copy the worker user data secret from `openshift-machine-api` namespace to `openshift-cluster-api`.
+
+At this point all Cluster API components should be installed and ready to use.
+
+#### CVO management
+
+A new `cluster-capi-operator` image will be built and included in every release payload.
+
+#### Credentials management
+
+The `cluster-capi-operator`'s manifests should contain an appropriate `CredentialsRequest` for each supported cloud provider.
+This is similiar to [machine-api-operator](https://github.com/openshift/machine-api-operator/blob/6f629682b791a6f4992b78218bfc6e41a32abbe9/install/0000_30_machine-api-operator_00_credentials-request.yaml)
+
+#### Cluster API cloud providers
+
+Cluster API cloud providers will live in a forks, similar to what is now done for Machine API.
+
+#### Example usage
+
+Usage is similar to Machine API with small differences, MachineSets reference the `Cluster` object and infrastructure machine template.
+
+```yaml
+---
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: MachineSet
+metadata:
+  name: capi-ms
+  namespace: openshift-cluster-api
+spec:
+  clusterName: cluster-name
+  replicas: 1
+  selector:
+    matchLabels: 
+      test: example
+  template:
+    metadata:
+      labels:
+        test: example
+    spec:
+      bootstrap:
+         dataSecretName: worker-user-data
+      clusterName: cluster-name
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+        kind: AWSMachineTemplate
+        name: cluster-name
+
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: AWSMachineTemplate
+metadata:
+  name: capi-machine-template
+  namespace: openshift-cluster-api
+spec:
+  template:
+    spec:
+      uncompressedUserData: true
+      iamInstanceProfile: ....
+      instanceType: m5.large
+      cloudInit:
+        secureSecretsBackend: secrets-manager
+        insecureSkipSecretsManager: true
+      ami:
+        id: ....
+      subnet:
+        filters:
+        - name: tag:Name
+          values:
+          - ...
+      additionalSecurityGroups:
+      - filters:
+        - name: tag:Name
+          values:
+          - ...
+```
+
+### Risks and Mitigations
+
+- During tech preview `cluster-capi-operator` will have permissions to manage CRDs, this might be a not secure permission for an operator.
+- Note, this permission should be restricted to creating CRDs only, as once installed, the technical preview cannot be uninstalled.
+- CLI usage, once Cluster API is installed command like `oc get machine` will return Cluster API machines, in order to use Machine API users will have to use fully qualified name `oc get machines.machine.openshift.io`.
```

Comment:
this makes me wonder if we shouldn't have some sort of warning message associated with `oc get machines` once the feature gate is active, but i'm not sure if that's even possible.

Reply:
Yeah, we still have to figure out what to do here.